### PR TITLE
test(e2e): backport protocol contract v2 migration test

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Part of https://github.com/zeta-chain/node/issues/2779

Backport https://github.com/zeta-chain/node/pull/2763 to v20

### CODE REVIEW

**Comment by lumtis** [comment]
> it seems that cmd/zetae2e/local/migration.go was removed in original PR but not here

This was a refactoring for code quality migration -> tss migration, not necessary here 

> and some not related changes in zetaclient/chains/bitcoin/observer/observer.go in original PR

refactoring to improve logs emitted in localnet, I deemed not crucial for the backport

File: x/fungible/keeper/msg_server_update_gateway_contract.go [hunk 1]
```diff
@@ -21,6 +23,16 @@ func (k msgServer) UpdateGatewayContract(
...
 
+	// parse the new gateway address
+	gatewayAddr := ethcommon.HexToAddress(msg.NewGatewayContractAddress)
+	if gatewayAddr == (ethcommon.Address{}) {
+		return nil, cosmoserrors.Wrapf(
+			sdkerrors.ErrInvalidAddress,
+			"invalid gateway contract address (%s)",
+			msg.NewGatewayContractAddress,
+		)
+	}
```
[the remark this answers is not in the document]
Comment (lumtis) [reply x/fungible/keeper/msg_server_update_gateway_contract.go:34]: This code backport changes on develop to the release, we should look into adding this change into `develop`, I've also opened a PR recently to refactor the empty address check

File: zetaclient/chains/evm/signer/signer.go [hunk 2]
```diff
@@ -115,9 +115,10 @@ func (signer *Signer) SetERC20CustodyAddress(addr ethcommon.Address) {
 }
 
 // SetGatewayAddress sets the gateway address
-func (signer *Signer) SetGatewayAddress(_ string) {
-	// Note: do nothing for now
-	// gateway address will be needed in the future contract architecture
+func (signer *Signer) SetGatewayAddress(addr string) {
+	signer.Lock()
+	defer signer.Unlock()
+	signer.gatewayAddress = ethcommon.HexToAddress(addr)
```
[the remark this answers is not in the document]
Comment (lumtis) [reply zetaclient/chains/evm/signer/signer.go:121]: The address will be zeroed if invalid, I think implementing a error handling mechanism would be bigger than the scope of the backport

**Review by fbac — APPROVED** [review]

**Review by gartnera — APPROVED** [review]

**Review by julianrubino — APPROVED** [review]
lgtm

**Pull request merged**